airflow-core.yaml

```diff
@@ -0,0 +1,136 @@
+package:
+  name: airflow-core
+  version: 2.10.4
+  epoch: 0
+  description: Minimal Airflow core - Platform to programmatically author, schedule, and monitor workflows
+  dependencies:
+    runtime:
+      - bash
+      - busybox
+      - coreutils
+      - curl
+      - cyrus-sasl
+      - cyrus-sasl-dev
+      - findutils
+      - glibc-locale-en
+      - glibc-locales
+      - gnupg-utils
+      - gzip
+      - mariadb-dev
+      - mysql
+      - openssl
+      - postgresql-client
+      - postgresql-dev
+      - python-3.12
+  copyright:
+    - license: Apache-2.0
+
+environment:
+  contents:
+    packages:
+      - gcc
+      - glibc-dev
+      - mariadb-connector-c-dev
+      - mariadb-dev
+      - openssf-compiler-options
+      - pkgconf-dev
+      - postgresql-dev
+      - py3.12-build-bin
+      - py3.12-pip
+      - python-3.12
+      - python-3.12-dev
+      - wolfi-base
+      - zlib-dev
+
+pipeline:
+  - uses: git-checkout
+    with:
+      repository: https://github.com/apache/airflow
+      tag: ${{package.version}}
+      expected-commit: c083e456fa02c6cb32cdbe0c9ed3c3b2380beccd
+
+  - runs: |
+      # by default airflow celery provider is not built,but running the upstream helm chart requires it
+       python patch.py
+       # remove the patcher
+       rm -rf patch.py
+
+  - runs: |
+      # Set required build environment variables
+      export SOURCE_DATE_EPOCH=315532800
+      export MYSQLCLIENT_CFLAGS=$(mysql_config --cflags)
+
+      # Install airflow core, the upstream sets this as directory
+      export PYTHONUSERBASE=${{targets.destdir}}/home/airflow/.local
+      pip3.12 install --no-cache-dir --user apache-airflow[core]==${{package.version}}
+
+      pip3.12 install --no-cache-dir --user aiohttp>=3.10.11 tornado>=6.4.2
+
+      pip3.12 install --no-cache-dir --user --force-reinstall packaging statsd
+
+  - runs: |
+      # Cleanup unnecessary dependencies
+      pip3.12 uninstall --yes setuptools
+      find . -name '__pycache__' -exec rm -rf {} +
+
+  - runs: |
+      mkdir -p ${{targets.destdir}}/opt/airflow/dags
+      mkdir -p ${{targets.destdir}}/opt/airflow/logs
+      mkdir -p ${{targets.destdir}}/scripts/docker
+
+      # The first time you run Airflow, it will create a file called `airflow.cfg` in
+      # `$AIRFLOW_HOME` directory
+      # However, for production case it is advised to generate the configuration
+      export PYTHONPATH=${{targets.destdir}}/home/airflow/.local/lib/python3.12/site-packages
+
+      ${{targets.destdir}}/home/airflow/.local/bin/airflow config list --defaults > ${{targets.destdir}}/opt/airflow/"airflow.cfg"
+
+      cp airflow/config_templates/default_webserver_config.py ${{targets.destdir}}/
+
+      cp scripts/docker/entrypoint_prod.sh ${{targets.destdir}}/entrypoint
+      chmod 755 ${{targets.destdir}}/entrypoint
```

Comment:
I don't believe permissions are persisted in melange, this would need to be done at the image level, so can remove this line

(same for similar references below)

Reply:
Actually the reference comes from here as airflow did that so I was not sure if we had some historical context to do it, we might have to refactor this too https://github.com/wolfi-dev/os/blob/66c102ad5a2b5defd16ce394e330b498efbaff4f/airflow.yaml#L113

Reply:
From the discussion with Srishti and comments from this issue by RJ, I think we are good to go with this to align with what our full blown airflow package does

https://github.com/chainguard-dev/internal-dev/issues/8158#issuecomment-2580349351

Reply:
Let's always handle in the package when we can, this works fine